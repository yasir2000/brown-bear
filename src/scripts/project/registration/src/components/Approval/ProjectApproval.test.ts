/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 *  This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 *
 */

import { shallowMount } from "@vue/test-utils";
import { createProjectRegistrationLocalVue } from "../../helpers/local-vue-for-tests";
import ProjectApproval from "./ProjectApproval.vue";
import { createStoreMock } from "../../../../../vue-components/store-wrapper-jest";
import VueRouter from "vue-router";

describe("ProjectApproval -", () => {
    let router: VueRouter;
    beforeEach(() => {
        router = new VueRouter({
            routes: [
                {
                    path: "/new",
                    name: "template",
                },
            ],
        });
    });
    it("Spawns the ProjectApproval component", async () => {
        const getters = {
            has_error: false,
            is_template_selected: true,
        };

        const store_options = {
            getters,
        };

        const store = createStoreMock(store_options);

        const wrapper = shallowMount(ProjectApproval, {
            localVue: await createProjectRegistrationLocalVue(),
            mocks: { $store: store },
            router,
        });

        expect(wrapper).toMatchSnapshot();
    });

    it("redirects user on /new when he does not have all needed information to start his project creation", async () => {
        const getters = {
            has_error: false,
            is_template_selected: false,
        };

        const store = createStoreMock({ getters });
        const wrapper = shallowMount(ProjectApproval, {
            localVue: await createProjectRegistrationLocalVue(),
            mocks: { $store: store },
            router,
        });
        expect(wrapper.vm.$route.name).toBe("template");
    });
});
