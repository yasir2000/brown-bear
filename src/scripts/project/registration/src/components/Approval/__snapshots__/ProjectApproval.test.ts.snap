// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectApproval - Spawns the ProjectApproval component 1`] = `
<div class="empty-state-page">
  <div class="empty-state-illustration">
    <project-approval-svg-stub></project-approval-svg-stub>
  </div>
  <h1 class="empty-state-title" data-msgid="
        Your project has been submitted to the administrators for validation
    " data-current-language="en_US">
    Your project has been submitted to the administrators for validation
  </h1>
  <p class="empty-state-text"><b>You will receive an email</b> when the administrator has validated it.</p> <a href="/my/" class="tlp-button-primary tlp-button-large empty-state-action"><i class="fa fa-reply tlp-button-icon"></i> <span data-msgid="Go to my home page" data-current-language="en_US">Go to my home page</span></a>
</div>
`;
